### ISSUE TITLE
When moving the "pop-out" settings window into a tab, both this window and the rule entry created for the highlighted word are duplicated.

### ISSUE THREAD

**Issue by xSwagboy69x** [fastaddons/AutoHighlight#30]
**Describe the bug**  

**ISSUES:** (3 separate ones)

1. The "pop-out" settings window (the one that appears after highlighting a word and selecting a context-menu option, _**not**_ the "pop-up" from the toolbar icon) has an arrow at the top-right corner that allows you to move it to a tab. When clicked, the pop-out window is cloned to a tab instead of being moved itself, which means there are now 2 settings windows (the original pop-out (still on the original page), and the newly created tab) and 2 new rule entries created. 

2. When the "Always open in new tab - when clicking toolbar icon" setting is *enabled*, clicking the toolbar icon does nothing. It does not open settings in a new tab, nor does it show the toolbar pop-up. Disabling the checkbox for this option immediately allows the toolbar pop-up to appear.

3. When selecting the "Add to existing rule" context-menu option, only 3-4 seemingly random choices appear. The available choices don't appear to have anything to do with the current website, and it's definitely not chronological. Could this be fixed to show all available rules that already exist for the current website?

**Steps To Reproduce**  
1. Highlight a word, and right-click to bring up context-menu.
2. Make a selection from the "Auto Highlight" submenu.
3. In the "pop-out" window that now shows, click on arrow at the top-right corner.

**Expected behavior**  
Upon clicking on the arrow, I'd expect the original settings pop-out to itself be moved to a tab, rather than a new instance of it cloned to tab. Also, I would not expect a second identical rule entry being created. Any rule changes / options selected in either window will also immediately show up in the other, in case that matters.

Presumably, the normal/intended behavior would for the original settings pop-out window itself be moved to a tab, or at least closed when the new tab is created. Also, there should not be 2 new rule entries created.

**Device info:**
 - operating system: Windows 11
 - Browser and version: Microsoft Edge (v117.0.2045.31 (Official build) (64-bit))
 - Auto Highlight version: 5.1 (installed from the official 'Edge Addons' site)

**Additional info**
I wasn't sure if I was supposed to create 3 new bug reports; I went with 1 because I assumed it would be less spammy and the 2nd/3rd problem(s) are minor. I apologize if there's an issue with this.

**Comment by Juraj-Masiar** [comment]
Hello,
Thank you for the bug reports!
It's OK to have them all here, I'll try to fix them in the upcoming release (this may take a two or more weeks since I'm in the middle of a big feature for my other extension).

Regarding the number 3, that one is the most tricky. Currently it shows all rules that have "each word" option active (so that the new word can be added there).
However, since context menu is globally set, restricting it for current page would mean I would have to recompute it every time you visit a page, plus when you switch tabs or active windows. It's not impossible, it's just more "resource consuming" and the trade-off doesn't feel right.
But maybe I can think of some workaround. There is actually one for Firefox which can [refresh already opened menu ](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/menus/refresh)(so I could recompute it when context menu is opened).
In other browsers I could maybe watch for right mouse button being pressed and recompute it there.
I'll see what I can do :)

**Comment by xSwagboy69x** [comment]
Hi, thanks for the reply! 

The "each word" option requirement makes sense – now I realize that I'm only seeing a handful of choices because virtually all of my rules are using a simple regex. Makes sense! It's a minor thing so no big deal.

As an aside, would it be possible for some form or organization for rules? Like grouping by site, or nested entries by site, something along those lines? This has quickly become my most used extension and it's becoming a bit difficult to keep track of all my rules :)

Anyhow, thanks for a great extension, it's truly useful!

**Comment by Juraj-Masiar** [comment]
Yeah, grouping is planned feature for some time now. It's bothering me a lot too.
I was waiting for my framework to add support for that but there was some issue.
The good news is that I found another way to add grouping (although maybe only one level deep) so I'll try to implement it "soon".
The bad news I have almost no time for this extension right now. But I'll try to allocate some time after I finish the big task I'm working on now.

**Comment by xSwagboy69x** [comment]
Even one level deep would be great, it's enough to reduce the visual overload and would make it much easier to visually find certain rules.

In fact, I'd even prefer getting that feature over any of the issues I mentioned in my original bug report! Hopefully it's not too complicated and can be done. I've been suggesting this extension to friends and everyone agrees how useful it is, I really hope you continue to develop it :)
